linalg/hypre.cpp

```diff
@@ -526,7 +526,7 @@ real_t ParNormlp(const Vector &vec, real_t p, MPI_Comm comm)
     this function. In particular, @a dst should be empty or deleted before
     calling this function. */
 template <typename T>
-void CopyMemory(Memory<T> &src, Memory<T> &dst, MemoryClass dst_mc,
+void CopyMemory(const Memory<T> &src, Memory<T> &dst, MemoryClass dst_mc,
```

Comment:
Adding `const` here can be deceiving -- it does not protect the content of the Memory -- note that the call `dst.MakeAlias` is a "shallow copy", so the contents of `src` is "exposed" in `dst`.

Reply:
Reverted in https://github.com/mfem/mfem/pull/4659/commits/193f8a68012d12772c179d01ba85df2dfedcb2e0 .